### PR TITLE
cam6_3_102: Fix cam_dev TEM diags issue

### LINKED ISSUE #769: TEM diags not implemented in cam_dev physics
### What happened?

The calls to `phys_grid_ctem` are not in `src/physics/cam_dev/physpkg.F90`, only in `src/physics/cam/physpkg.F90`.  

### What are the steps to reproduce the bug?

Use configure option `-phys cam_dev`

and add namelist settings to turn on TEM diags, such as:
```
  phys_grid_ctem_nfreq=-6
  phys_grid_ctem_zm_nbas=120 
  phys_grid_ctem_za_nlat=90
  fincl3 = 'VTHzaphys','WTHzaphys','UVzaphys','UWzaphys'
```

### What CAM tag were you using?

cam6_3_100

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

Yes

### Extra info

_No response_

### PR DESCRIPTION
**Pull request by fvitt**
fixes #769 

Also :
 - adopts the `zm` naming convention for the zonal mean outputs.  
 - add 'Uzm','Vzm','Wzm','THzm' output fields

### CODE REVIEW

**Comment by fvitt** [comment]
@cacraigucar This is ready for full review now.  Thanks.

File: src/physics/cam/phys_grid_ctem.F90 [hunk 1]
```diff
@@ -140,7 +140,7 @@ subroutine phys_grid_ctem_reg
     real(r8), parameter :: latrad0 = -pi*0.5_r8
     real(r8), parameter :: fourpi = pi*4._r8
 
-    integer, parameter :: ctem_zavg_phys_decomp = 201 ! Must be unique within CAM
+    integer, parameter :: ctem_zavg_phys_decomp = 333 ! Must be unique within CAM
```
Comment (cacraigucar) [inline src/physics/cam/phys_grid_ctem.F90:143]: Just curious what this number is and why it changed.  Perhaps a little bit more description in the comment would be helpful?

**Review by cacraigucar — APPROVED** [review]

File: src/physics/cam/phys_grid_ctem.F90 (hunk 1 above)
Comment (fvitt) [reply src/physics/cam/phys_grid_ctem.F90:143]: The number is used for registration of the zonal mean history grid.  The 201 number apparently was not unique.  There was a problem when I tried to use SE grid interpolate_output feature and TEM zm output in the same run.  Changing the number fixed the issue.

File: src/physics/cam/phys_grid_ctem.F90 [hunk 2]
```diff
@@ -207,11 +207,15 @@ subroutine phys_grid_ctem_init
...
 
-    call addfld ('VTHzaphys',(/'lev'/), 'A', 'K m s-1','Meridional Heat Flux:', gridname='ctem_zavg_phys')
-    call addfld ('WTHzaphys',(/'lev'/), 'A', 'K m s-1','Vertical Heat Flux:', gridname='ctem_zavg_phys')
-    call addfld ('UVzaphys', (/'lev'/), 'A', 'm2 s-2', 'Meridional Flux of Zonal Momentum', gridname='ctem_zavg_phys')
-    call addfld ('UWzaphys', (/'lev'/), 'A', 'm2 s-2', 'Vertical Flux of Zonal Momentum', gridname='ctem_zavg_phys')
-    call addfld ('THphys',   (/'lev'/), 'A', 'K',      'Potential temp', gridname='physgrid' )
+    call addfld ('Uzm',  (/'lev'/), 'A','m s-1',  'Zonal-Mean zonal wind - defined on ilev', gridname='ctem_zavg_phys' )
+    call addfld ('Vzm',  (/'lev'/), 'A','m s-1',  'Zonal-Mean meridional wind - defined on ilev', gridname='ctem_zavg_phys' )
+    call addfld ('Wzm',  (/'lev'/), 'A','m s-1',  'Zonal-Mean vertical wind - defined on ilev', gridname='ctem_zavg_phys' )
+    call addfld ('THzm', (/'lev'/), 'A','K',      'Zonal-Mean potential temp - defined on ilev', gridname='ctem_zavg_phys' )
```
Comment (nusbaume) [inline src/physics/cam/phys_grid_ctem.F90:213]: It looks like these are all defined on `lev`, not `ilev`.  Should we remove the `ilev` statement from the history field description?

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good, just one request to update some history field descriptions.

File: src/physics/cam/phys_grid_ctem.F90 (hunk 2 above)
Comment (fvitt) [reply src/physics/cam/phys_grid_ctem.F90]: fixed

**Review by nusbaume — APPROVED** [review]
Everything looks good to me now.  Thanks!

**Review by brian-eaton — APPROVED** [review]

**Pull request merged**